src/Libraries/CoreNodeModels/Selection.cs

```diff
@@ -102,8 +102,12 @@ public virtual string SelectionSuggestion
 
             this.selectionType = selectionType;
             this.selectionObjectType = selectionObjectType;
-
-            OutPortData.Add(new PortData("Elements", Resources.SelectionPortDataResultToolTip));
+            if (selectionMessage.Equals("Select Model Element"))
+            {
+                OutPortData.Add(new PortData("Element", Resources.SelectionPortDataResultToolTip));
+            }
+            else 
+                OutPortData.Add(new PortData("Elements", Resources.SelectionPortDataResultToolTip));
```

Comment:
1) I didn't see any difference between if and else statement
2) what does `selectionMessage` come from? Compare with hard coded string is not reliable.